src/hooks/useI18n.js

```diff
@@ -1,4 +1,30 @@
 import { useContext } from 'preact/hooks'
 import { I18nContext } from 'contexts'
+import { getAppLanguage } from 'utils'
 
-export const useI18n = () => useContext(I18nContext)
+export const useI18n = lang => {
+  const banana = useContext(I18nContext)
+
+  return (key, ...args) => {
+    if (lang) {
+      const locale = banana.locale
+      let msg
+      try {
+        banana.setLocale(lang)
+      } finally {
+        msg = banana.i18n(key, ...args)
```

Comment:
I think this line should be in the `try` block

Reply:
if this line in `try` block, msg will be undefined, we need to get it as least the fallback lang

Reply:
This is not related to fallback. The try/finally is to ensure that if `banana.i18n(...)` throws an exception, the locale in the shared instance is reset to what it should be.

However, re-reading the code, I think the logic can be simplified to something like
```
lang = lang || getAppLanguage()
if (banana.locale !== lang) {
  banana.setLocale(lang)
}
return return banana.i18n(key, ...args)
```

The idea is that every time the function is called, it sets the right locale (if it needs to). Do you see any issue with this approach?

Reply:
it totally makes sense and the code much simpler now

what benefit actually do we get in try/catch form when setLocale? identify the language we are not supported yet?